File: frontend/src/types/chore.ts
export interface Chore {
  id: string;
  description: string;
  assigned_to: string;
  completed: boolean;
  created_at: string;
  updated_at: string;
}

export interface ChoreWithUser {
  id: string;
  description: string;
  assigned_to: string;
  assigned_name: string;
  completed: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateChoreInput {
  description: string;
  assigned_to: string;
}

export interface UpdateChoreInput {
  description?: string;
  assigned_to?: string;
  completed?: boolean;
}

File: backend/src/models/chore.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct Chore {
    pub id: Uuid,
    pub description: String,
    pub assigned_to: Option<Uuid>,
    pub reward: Option<i64>,
    pub completed: bool,
    pub created_at: chrono::DateTime<chrono::Utc>,
    pub updated_at: chrono::DateTime<chrono::Utc>,
}

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct ChoreWithUser {
    pub id: Uuid,
    pub description: String,
    pub assigned_to: Option<Uuid>,
    pub assigned_name: Option<String>,
    pub reward: Option<i64>,
    pub completed: bool,
    pub created_at: chrono::DateTime<chrono::Utc>,
    pub updated_at: chrono::DateTime<chrono::Utc>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct CreateChoreSchema {
    pub description: String,
    pub assigned_to: Option<Uuid>,
    pub reward: Option<i64>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct UpdateChoreSchema {
    pub description: Option<String>,
    pub assigned_to: Option<Uuid>,
    pub reward: Option<i64>,
    pub completed: Option<bool>,
}
